overlord/configstate/configcore/watchdog_test.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type watchdogSuite struct {
+	configcoreSuite
+
+	mockEtcEnvironment string
+}
+
+var _ = Suite(&watchdogSuite{})
+
+func (s *watchdogSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	dirs.SetRootDir(c.MkDir())
+	err := os.MkdirAll(filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/"), 0755)
+	c.Assert(err, IsNil)
+	s.mockEtcEnvironment = filepath.Join(dirs.GlobalRootDir,
+		"/etc/systemd/system.conf.d/10-watchdog.conf")
+}
+
+func (s *watchdogSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+func (s *watchdogSuite) makeMockEtcEnvironment(c *C) {
+	err := ioutil.WriteFile(s.mockEtcEnvironment, []byte{}, 0644)
+	c.Assert(err, IsNil)
+}
+
+func (s *watchdogSuite) TestConfigureWatchdog(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	for option, val := range map[string]string{"runtime": "10", "shutdown": "60"} {
+		// populate with content
+		s.makeMockEtcEnvironment(c)
+
+		err := configcore.Run(&mockConf{
+			state: s.state,
+			conf: map[string]interface{}{
+				fmt.Sprintf("watchdog.%s", option): val,
+			},
+		})
+		c.Assert(err, IsNil)
+
+		var systemdOption string
+		switch option {
+		case "runtime":
+			systemdOption = "RuntimeWatchdogSec"
+		case "shutdown":
+			systemdOption = "ShutdownWatchdogSec"
+		}
+		c.Check(s.mockEtcEnvironment, testutil.FileEquals,
+			fmt.Sprintf("[Manager]\n%s=%s\n", systemdOption, val))
+	}
+}
+
+func (s *watchdogSuite) TestConfigureWatchdogAll(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	// populate with content
+	s.makeMockEtcEnvironment(c)
+
+	times := []int{10, 100}
+	err := configcore.Run(&mockConf{
+		state: s.state,
+		conf: map[string]interface{}{
+			"watchdog.runtime":  times[0],
+			"watchdog.shutdown": times[1],
+		},
+	})
+	c.Assert(err, IsNil)
+	c.Check(s.mockEtcEnvironment, testutil.FileContains, "[Manager]\n")
```

Comment:
I think its slightly nicer to check the full content here, like:
```
c.Check(s.mockEtcEnvironment, testutil.FileEquals, fmt.Sprintf(`[Manager]
RuntimeWatchdogSec=%d
ShutdownWatchdogSec=%d
`, times[0], times[1]))
```
this way we test that the file looks exactly what we want it to look. The other test may hide extra lines that somehow creped in.

Reply:
Right, but the code uses a map and the order of the options is not deterministic, so this cannot be done directly like you say.